### ISSUE TITLE
Auto Move between Text Fields not working

### ISSUE THREAD

**Issue by ChrisLace** [codenameone/CodenameOne#2459]
Issue: When the Text field auto move functionality doesn't work as per the developer guide on both android and ios devices. This is adding flickering effect on the textfield and misalignment for value entered in text field. Please create a sample case with following code and you should  be able to generate the error. This error doesn't occur on simulator.

This issue has been discussed already. Please refer the ticket https://stackoverflow.com/questions/49742976/automovetonext-method-from-developer-guide-is-not-working-properly.

I have just create one test case to replicate the issue. Same thing happens if we tried to do US Phone Number control using 4 text fields in the format 111-222-3333-4444.

Test Case:
 
````java
TextField num1 = new TextField("", "Phone Number", 10, TextArea.NUMERIC);
TextField num2 = new TextField("", "Extension", 4, TextArea.NUMERIC);

onTypeNext(num1, num2, 10);

private void onTypeNext(TextField current, TextField next, int length) {
    current.addDataChangedListener((i, ii) -> {
        if(current.getText().length() == length) {
            current.stopEditing();
            next.startEditingAsync();
        }
    });
}
````

**Comment by codenameone** [comment]
I edited the issue to fix the code listing. Please specify the device and clarify the issue. Linking to a stackoverflow issue with a long discussion isn't a good practice because it's hard to follow and stackoverflow sometimes deletes issues. 
Please clarify what isn't working and if applicable attach screenshots.

**Comment by ChrisLace** [comment]
I have added more description for the issue. Please let me know if any questions.

**Comment by ChrisLace** [comment]
![error](https://user-images.githubusercontent.com/39275611/41814280-dc7c5ee0-76fc-11e8-92c9-d30c914ad35a.JPG)

**Comment by ChrisLace** [comment]
I have found this also happening with text fields. I thought this is related one. Please see this
![textfield](https://user-images.githubusercontent.com/39275611/42003373-0180770a-7a20-11e8-99a5-80f1914709f4.jpg)
